Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/CounterDefinitionSample.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <summary>
+    /// Represents performance data for an individual counter.
+    /// </summary>
+    internal class CounterDefinitionSample
+    {
+        internal readonly int NameIndex;
+
+        internal readonly int CounterType;
+
+        internal long[] InstanceValues;
+
+        private readonly int size;
+
+        private readonly int offset;
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CounterDefinitionSample"/> class. 
+        /// </summary>
+        /// <param name="perfCounter">Performance counter data.</param>
+        /// <param name="instanceNumber">Instance name.</param>
+        public CounterDefinitionSample(NativeMethods.PERF_COUNTER_DEFINITION perfCounter, int instanceNumber)
+        {
+            this.NameIndex = perfCounter.CounterNameTitleIndex;
+            this.CounterType = perfCounter.CounterType;
+            this.offset = perfCounter.CounterOffset;
+            this.size = perfCounter.CounterSize;
+            this.InstanceValues = instanceNumber == -1 ? new long[1] : new long[instanceNumber];
+        }
+
+        /// <summary>
+        /// Sets the value for an instance.
+        /// </summary>
+        /// <param name="index">Index value.</param>
```

Comment:
Some of these could be more meaningful but thank-you for at least putting something.